docs/guides/rule-evaluations-via-query-frontend.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Rule evaluations via query frontend"
+linkTitle: "Rule evaluations via query frontend"
+weight: 10
+slug: rule-evalutions-via-query-frontend
+---
+
+This guide explains how to configure the Ruler to evaluate rules via Query Frontends instead of the Ingesters and the pros and cons of rule evaluation via Query Frontend.
+
+## How to enable
+
+By default, the Ruler queries to the Ingesters for evaluating rules (alerting rules or recording rules). If you have set `-ruler.frontend-address` then the Ruler query to the Query Frontend for evaluation rules.
+The address should be the gRPC listen address in host:port format.
+
+You can configure via args:
+```
+-ruler.frontend-address=query-frontend.svc.cluster.local:9095
+```
+And via yaml:
+```yaml
+ruler:
+  frontend_address: query-frontend.svc.cluster.local:9095
+```
+
+In addition, you can configure gRPC client (Ruler -> Query Frontend) config via args:
+```
+-ruler.frontendClient.grpc-max-recv-msg-size=104857600
+-ruler.frontendClient.grpc-max-send-msg-size=16777216
+-ruler.frontendClient.grpc-compression=""
+-ruler.frontendClient.grpc-client-rate-limit=0
+-ruler.frontendClient.grpc-client-rate-limit-burst=0
+-ruler.frontendClient.backoff-on-ratelimits=false
+-ruler.frontendClient.backoff-min-period=100ms
+-ruler.frontendClient.backoff-max-period=10s
+-ruler.frontendClient.backoff-retries=10
+```
+
+And via yaml:
+
+```yaml
+ruler:
+  frontend_client:
+    max_recv_msg_size: 104857600
+    max_send_msg_size: 16777216
+    grpc_compression: ""
+    rate_limit: 0
+    rate_limit_burst: <int> | default = 0
+    backoff_on_ratelimits: <boolean> | default = false
+    backoff_config:
+      min_period: 100ms
+      max_period: 10s
+      max_retries: 10
+```
+
+## Pros and Cons
+
+### Pros
+The rule evaluation performance is improved as we can use Query Frontend features like the vertical query sharding.
```

Comment:
In terms of performance, I think it is complex and it is not always true to say evaluation performance is improved.

By evaluating rules through Query Frontend, there are more hops in this path. Ruler -> Query Frontend -> Query Scheduler -> Querier -> Ingester/Store Gateway.

Query Frontend/Query Scheduler has a query queue and Querier pulls queries from the queue to execute. If there are not enough queriers, add rule queries to this path could cause query starvation problem and rules cannot get evaluated. I think this is something to call out in this doc.